### PR TITLE
`Development`: Rename plagiarism-post related classes

### PR DESCRIPTION
**Pull request by lennart-keller**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).(https://docs.artemis.cit.tum.de/dev/guidelines/server/#rest-endpoint-best-practices-for-authorization) and checked the course groups for all new REST Calls (security).
- [x] I documented the Java code using JavaDoc style.

### Motivation and Context

Classes that are related to plagiarism case posts have are difficult to identify for developers.

### Description

Plagiarism-post related classes are now prefixed with `Plagiarism`. Classes have also been moved to the package for plagiarism cases.

Methods in the `PostSpecs.java` file are only used for conversation messages. The class has been removed and the methods have been moved to `MessageSpecs.java`

### Steps for Testing

Prerequisites:
- 1 Student

1. Go to a course with communication enabled
2. Try out the search functionality on the communication tab
3. Send a message and a reply on the messages tab
4. React to a message and a reply

#### Performance Review
- [x] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage

Same as develop

### Screenshots

No UI changes

## Summary by CodeRabbit

- **New Features**
  - Enhanced search functionality in conversations with new criteria-based fetching.

- **Improvements**
  - Improved reaction handling with updated services for better performance.
  - Streamlined conversation-related imports for more efficient code maintenance.
  - Refined plagiarism detection services for more accurate case handling.

- **Refactor**
  - Renamed and reorganized packages and services related to plagiarism features.
  - Consolidated import statements across various classes for cleaner code.

- **Tests**
  - Updated test suites to reflect new service structure and naming conventions.
  - Introduced wildcard imports in test classes for simplified code.

### CODE REVIEW

**Review by basak-akan — DISMISSED** [review]
Code lgtm. Also tested filtering options on the communication tab and the lecture and exercise detail pages, works well.

**Review by basak-akan — APPROVED** [review]
Re-approve, thank you for addressing the numerical search

**Review by milljoniaer — APPROVED** [review]
Code LGTM 🚀

**Review by egekurt123 — DISMISSED** [review]
Tested manually on ts5, works as expected

**Review by georgyia — APPROVED** [review]
manually tested on ts2, works as expected

**Review by rstief — APPROVED** [review]
Code LGTM

**Comment by konrad2002** [comment]
manually tested on legacy ts11, works as expected

**Review by konrad2002 — DISMISSED** [review]
manually tested on legacy ts11, works as expected

**Review by rriyaldhi — DISMISSED** [review]
Code change looks good

File: src/test/java/de/tum/in/www1/artemis/service/DataExportCreationServiceTest.java [hunk 1]
```diff
@@ -10,15 +10,14 @@
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.time.ZonedDateTime;
-import java.util.ArrayList;
-import java.util.Comparator;
-import java.util.List;
-import java.util.Set;
+import java.util.*;
```
[the remark this answers is not in the document]
Comment (lennart-keller) [reply src/test/java/de/tum/in/www1/artemis/service/DataExportCreationServiceTest.java:13]: We only allow 4 imports or use * instead

File: src/main/java/de/tum/in/www1/artemis/service/plagiarism/PlagiarismPostService.java [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package de.tum.in.www1.artemis.service.metis;
+package de.tum.in.www1.artemis.service.plagiarism;
 
 import java.time.ZonedDateTime;
 import java.util.List;
```
[the remark this answers is not in the document]
Comment (lennart-keller) [reply src/main/java/de/tum/in/www1/artemis/service/plagiarism/PlagiarismPostService.java:4]: The entity itself is a post, so the naming is ok

**Review by rriyaldhi — APPROVED** [review]
Reapprove after fixing the logger class names

**Review by rstief — APPROVED** [review]
code looks good after changes.

**Review by basak-akan — APPROVED** [review]
Re-tested on TS1, still works as expected

**Review by milljoniaer — APPROVED** [review]
reapprove after code changes 👍

**Review by xHadie — APPROVED** [review]
_Manually tested on TS11_Legacy_
Works as expected
image

**Pull request merged**